Extensions/Firebase/tests/FirebaseExtension.spec.js

```diff
@@ -0,0 +1,459 @@
+            gdjs.evtTools.firebaseTools.firestore.hasDocument(
+              namespace,
+              'delete',
+              result,
+              callback
+            )
+          )
+        ).getValue()
+      ).to.be.ok();
+
+      expect(
+        (
+          await promisifyCallbackVariables((callback, result) =>
+            gdjs.evtTools.firebaseTools.firestore.hasField(
+              namespace,
+              'delete',
+              'hello',
+              result,
+              callback
+            )
+          )
+        ).getValue()
+      ).to.be.ok();
+
+      await promisifyCallbackVariables((callback) =>
+        gdjs.evtTools.firebaseTools.firestore.deleteField(
+          namespace,
+          'delete',
+          'hello',
+          callback
+        )
+      );
+
+      expect(
+        (
+          await promisifyCallbackVariables((callback, result) =>
+            gdjs.evtTools.firebaseTools.firestore.hasField(
+              namespace,
+              'delete',
+              'hello',
+              result,
+              callback
+            )
+          )
+        ).getValue()
+      ).to.not.be.ok();
+
+      await promisifyCallbackVariables((callback) =>
+        gdjs.evtTools.firebaseTools.firestore.deleteDocument(
+          namespace,
+          'delete',
+          callback
+        )
+      );
+
+      expect(
+        (
+          await promisifyCallbackVariables((callback, result) =>
+            gdjs.evtTools.firebaseTools.firestore.hasDocument(
+              namespace,
+              'delete',
+              result,
+              callback
+            )
+          )
+        ).getValue()
+      ).to.not.be.ok();
+    });
+
+    // Delete the temporary namespace to not bloat the DB
+    after(async () =>
+      (
+        await firebase.firestore().collection(namespace).get()
+      ).forEach(({ ref }) => ref.delete())
+    );
+  });
+
+  describe('Firebase authentication', () => {
+    before(async () => firebase.auth().signOut());
+
+    it('Prevents logging in with invalid credentials', async () => {
+      expect(gdjs.evtTools.firebaseTools.auth.isAuthentified()).to.not.be.ok();
+
+      let errors = false;
+      try {
+        await promisifyCallbackVariables((callback) =>
+          gdjs.evtTools.firebaseTools.auth.signInWithEmail(
+            'yes@example.com',
+            'InvalidPassword321',
+            callback
+          )
+        );
+      } catch (e) {
+        errors = true;
+      }
+
+      // No error was throwm, there is an issue
+      if (!errors)
+        throw new Error('Expected wrong credentials to prevent login');
+
+      expect(gdjs.evtTools.firebaseTools.auth.isAuthentified()).to.not.be.ok();
+    });
+
+    it('Let users log in', async () => {
+      expect(gdjs.evtTools.firebaseTools.auth.isAuthentified()).to.not.be.ok();
+      await promisifyCallbackVariables((callback) =>
+        gdjs.evtTools.firebaseTools.auth.signInWithEmail(
+          'yes@example.com',
```

Comment:
I am concerned about those hardcoded credentials: Firebase doesn't allow disabling password resetting afaik, so someone could probably change the password and break tests :/